packages/visx-xychart/src/components/series/private/BaseAreaStack.tsx

```diff
@@ -0,0 +1,259 @@
+  }, [xScale, yScale, horizontal]);
+
+  // pull out all area + line props for each dataKey
+  const stacks = useMemo(
+    () =>
+      stackedData.map((stack, stackIndex) => {
+        const areaSeries:
+          | React.ReactElement<BaseAreaSeriesProps<XScale, YScale, Datum>>
+          | undefined = seriesChildren.find(child => child.props.dataKey === stack.key);
+        const {
+          data,
+          dataKey,
+          xAccessor,
+          yAccessor,
+          curve: _,
+          PathComponent: __,
+          lineProps,
+          renderLine: ___,
+          ...svgPathProps
+        } = areaSeries?.props || {};
+
+        const areaProps: SVGProps<SVGPathElement> = {
+          fill: colorScale?.(stack.key) ?? theme?.colors?.[0] ?? '#222',
+          ...svgPathProps,
+        };
+
+        return {
+          key: `${stackIndex}-${stack.key}`,
+          accessors,
+          data: stack,
+          areaProps,
+          lineProps,
+        };
+      }),
+    [stackedData, accessors, colorScale, seriesChildren, theme],
+  );
+
+  // custom logic to find the nearest AreaStackDatum (context) and return the original Datum (props)
+  const findNearestDatum = useCallback(
+    (params: NearestDatumArgs<XScale, YScale, AreaStackDatum>): NearestDatumReturnType<Datum> => {
+      const childData = seriesChildren.find(child => child.props.dataKey === params.dataKey)?.props
+        ?.data;
+      return childData ? findNearestStackDatum(params, childData, horizontal) : null;
+    },
+    [seriesChildren, horizontal],
+  );
+
+  const ownEventSourceKey = `${AREASTACK_EVENT_SOURCE}-${dataKeys.join('-')}`;
+  const eventEmitters = useSeriesEvents<XScale, YScale, Datum>({
+    dataKey: dataKeys,
+    enableEvents,
+    // @ts-ignore Datum input + return type are expected to be the same type but they differ
+    // for AreaStack (registry data is StackedDatum, return type is user Datum)
+    findNearestDatum,
+    onBlur,
+    onFocus,
+    onPointerMove,
+    onPointerOut,
+    onPointerUp,
+    source: ownEventSourceKey,
+    allowedSources: [XYCHART_EVENT_SOURCE, ownEventSourceKey],
+  });
+
+  // render invisible glyphs for focusing if onFocus/onBlur are defined
+  const captureFocusEvents = Boolean(onFocus || onBlur);
+  const renderGlyphs = useCallback(
+    ({ glyphs }: GlyphsProps<XScale, YScale, AreaStackDatum>) =>
+      captureFocusEvents
+        ? glyphs.map(glyph => (
+            <React.Fragment key={glyph.key}>
+              {defaultRenderGlyph({
+                ...glyph,
+                color: 'transparent',
+                onFocus: eventEmitters.onFocus,
+                onBlur: eventEmitters.onBlur,
+              })}
+            </React.Fragment>
+          ))
+        : null,
+    [captureFocusEvents, eventEmitters.onFocus, eventEmitters.onBlur],
+  );
+
+  // if scales and data are not available in the registry, bail
+  if (dataKeys.some(key => dataRegistry.get(key) == null) || !xScale || !yScale || !colorScale) {
```

Comment:
perhaps move the `!xScale || !yScale || !colorScale` check before the `dataKeys.some` because they are cheaper. Although `dataKeys` is negligibly small.